txpool/src/containers/dependency.rs

```
@@ -0,0 +1,482 @@
+                        return Err(Error::NotInsertedIOWrongAmount.into());
+                    }
+                    if color != i_color {
+                        return Err(Error::NotInsertedIOWrongAssetIt.into());
+                    }
+                }
+                Output::Contract { .. } => return Err(Error::NotInsertedIOConractOutput.into()),
+                Output::Withdrawal { .. } => {
+                    return Err(Error::NotInsertedIOWithdrawalInput.into());
+                }
+                Output::Change { to, color, amount } => {
+                    if to != i_owner {
+                        return Err(Error::NotInsertedIOWrongOwner.into());
+                    }
+                    if color != i_color {
+                        return Err(Error::NotInsertedIOWrongAssetIt.into());
+                    }
+                    if is_output_filled && amount < i_amount {
+                        return Err(Error::NotInsertedIOWrongAmount.into());
+                    }
+                }
+                Output::Variable { to, amount, color } => {
+                    if is_output_filled {
+                        if to != i_owner {
+                            return Err(Error::NotInsertedIOWrongOwner.into());
+                        }
+                        if amount < i_amount {
+                            return Err(Error::NotInsertedIOWrongAmount.into());
+                        }
+                        if color != i_color {
+                            return Err(Error::NotInsertedIOWrongAssetIt.into());
+                        }
+                    }
+                    // else do nothing, everything is variable and can be only check on execution
+                }
+                Output::ContractCreated { .. } => {
+                    return Err(Error::NotInsertedIOConractOutput.into())
+                }
+            };
+        } else {
+            panic!("Use it only for coin output check");
```

Why panic instead of just returning a normal `Error`?

---

If this uses function named `check_if_coin_input_can_spend_output` is used for checking `coin_contract` than this means somebody is missusing it